### PR TITLE
Fix findRStr and findStr misunderstood descriptions

### LINKED ISSUE #1857: findRStr and findStr misunderstood
## Describe the bug
According the wiki page StrFind return the position of an string in another string, from left to right.
StrRFind should do the same job but from the right to the left, but return the same value as StrFind when the research term is once in the string.

There is a misunderstood with these expression because StrRFind return the lastest occurence of the research and start the count from the left.
People understand that the expression will return the position from the right.

[Wiki decription:](http://wiki.compilgames.net/doku.php/gdevelop5/all-features/expressions-reference#expressions_reference)
`
StrRFind(string, string) = Search in a text from the end (return the position of the result or -1 if not found)
`

Correct case presently in gd:

```
var domain="www.test.com";
var lastdot= domain.lastIndexOf("t");  //return 7, from left to the right
```
How people think have the value:
```
var domain="www.test.com";
var lastdot= Math.abs(domain.lastIndexOf("t") - domain.length+1); //return 4, from right to left
// it's always the lastest 't' but from the end like it's explain on the wiki.
```

So what to do ?
I know if i tweak the expression findRStr, this will be broken for all previous games.
So maybe there is a simplier way to find the position from the end, or i can add a new expression for it, and we edit the description of the findRStr for explain clearly what is expected.

## Other details
b97

### PR DESCRIPTION
**Pull request by Bouh**
Copy and rename the old expressions and improve their descriptions.
Set deprecated and hidden the previous one.

Fix #1857

### CODE REVIEW

File: GDCpp/GDCpp/Extensions/Builtin/StringInstructionsExtension.cpp [hunk 1]
```diff
@@ -46,11 +46,17 @@ StringInstructionsExtension::StringInstructionsExtension() {
   GetAllExpressions()["StrRFind"]
       .SetFunctionName("GDpriv::StringTools::StrRFind")
```
Comment (4ian) [inline GDCpp/GDCpp/Extensions/Builtin/StringInstructionsExtension.cpp:47]: Let's change this function from StrRFind to StrFindLast

File: GDCpp/GDCpp/Extensions/Builtin/StringInstructionsExtension.cpp [hunk 2]
```diff
@@ -46,11 +46,17 @@ StringInstructionsExtension::StringInstructionsExtension() {
...
       .SetIncludeFile("GDCpp/Extensions/Builtin/StringTools.h");
+  GetAllExpressions()["StrFindLast"]
+      .SetFunctionName("GDpriv::StringTools::StrFindLast")
+      .SetIncludeFile("GDCpp/Extensions/Builtin/StringTools.h");
   GetAllExpressions()["StrFindFrom"]
       .SetFunctionName("GDpriv::StringTools::StrFindFrom")
       .SetIncludeFile("GDCpp/Extensions/Builtin/StringTools.h");
   GetAllExpressions()["StrRFindFrom"]
       .SetFunctionName("GDpriv::StringTools::StrRFindFrom")
```
Comment (4ian) [inline GDCpp/GDCpp/Extensions/Builtin/StringInstructionsExtension.cpp:56]: Same, StrRFindFrom => StrFindLastFrom

File: GDCpp/GDCpp/Extensions/Builtin/StringTools.h [hunk 3]
```diff
@@ -26,12 +26,16 @@ gd::String GD_API StrRepeat(const gd::String& str, std::size_t repCount);
 std::size_t GD_API StrLen(const gd::String& str);
 int GD_API StrFind(const gd::String& str, const gd::String& findwhat);
 int GD_API StrRFind(const gd::String& str, const gd::String& findwhat);
+int GD_API StrFindLast(const gd::String& str, const gd::String& findwhat);
```
Comment (4ian) [inline GDCpp/GDCpp/Extensions/Builtin/StringTools.h:29]: This is just adding the declaration of a function ;) Instead, let's rename StrRFind to StrFindLast (and some for the other functions), both here and in StringTools.cpp

File: GDJS/GDJS/Extensions/Builtin/StringInstructionsExtension.cpp [hunk 4]
```diff
@@ -35,10 +35,14 @@ StringInstructionsExtension::StringInstructionsExtension() {
       "gdjs.evtTools.string.strFind");
   GetAllExpressions()["StrRFind"].SetFunctionName(
       "gdjs.evtTools.string.strRFind");
```
Comment (4ian) [inline GDJS/GDJS/Extensions/Builtin/StringInstructionsExtension.cpp:37]: Let's use `gdjs.evtTools.string.strFindLast`

File: GDJS/GDJS/Extensions/Builtin/StringInstructionsExtension.cpp [hunk 5]
```diff
@@ -35,10 +35,14 @@ StringInstructionsExtension::StringInstructionsExtension() {
...
+  GetAllExpressions()["StrFindLast"].SetFunctionName(
+      "gdjs.evtTools.string.strFindLast");
   GetAllExpressions()["StrFindFrom"].SetFunctionName(
       "gdjs.evtTools.string.strFindFrom");
   GetAllExpressions()["StrRFindFrom"].SetFunctionName(
       "gdjs.evtTools.string.strRFindFrom");
```
Comment (4ian) [inline GDJS/GDJS/Extensions/Builtin/StringInstructionsExtension.cpp:43]: Same as before, let's change this to `gdjs.evtTools.string.strFindLastFrom`

File: GDJS/Runtime/events-tools/stringtools.js [hunk 6]
```diff
@@ -88,33 +88,51 @@ gdjs.evtTools.string.strLen = function(str) {
...
 };
 
 /**
- * Reverse search in a string
+ * Search the last occurence in a string (return the position of the result, from the beginning of the string, or -1 if not found)
  * @private
+ * @deprecated
  */
 gdjs.evtTools.string.strRFind = function(str, what) {
     return str.lastIndexOf(what);
```
Comment (4ian) [inline GDJS/Runtime/events-tools/stringtools.js:104]: As it's deprecated, let's implement it in terms of the not deprecated method:
`return gdjs.evtTools.string.strFindLast(str, what);`

File: GDJS/Runtime/events-tools/stringtools.js [hunk 7]
```diff
@@ -88,33 +88,51 @@ gdjs.evtTools.string.strLen = function(str) {
...
 };
 
 /**
- * Reverse search in a string, starting from a specified position.
+ * Search the last occurence in a string, starting from a specified position (return the position of the result, from the beginning of the string, or -1 if not found)
  * @private
+ * @deprecated
  */
 gdjs.evtTools.string.strRFindFrom = function(str, what, pos) {
     return str.lastIndexOf(what, pos);
```
Comment (4ian) [inline GDJS/Runtime/events-tools/stringtools.js:129]: Same, let's call `gdjs.evtTools.string.strFindLastFrom` ;)

**Comment by 4ian** [comment]
Thanks 👍 In this situation I think it's a good idea to just rename the JS functions (games should not use them anyway - they are not part of the public interface, it's a detail of the implementation) and use it for both the new expression and the old one :)
So that existing games using them will continue to work, and new games will use the not hidden expression.

File: GDJS/Runtime/events-tools/stringtools.js (hunk 7 above)
Comment (4ian) [reply GDJS/Runtime/events-tools/stringtools.js:129]: Actually, we can just do: `gdjs.evtTools.string.strRFindFrom = gdjs.evtTools.string.strFindLastFrom`. Very clear and concise!

File: GDJS/Runtime/events-tools/stringtools.js (hunk 6 above)
Comment (4ian) [reply GDJS/Runtime/events-tools/stringtools.js:104]: Actually, `gdjs.evtTools.string.strRFind = gdjs.evtTools.string.strFindLast;` will do the trick ;)

**Comment by 4ian** [comment]
I went ahead and implemented the comment fixes by myself ;) Thanks for the PR!

**Comment by Bouh** [comment]
Yes no worries i was just impossible to me to do the fixes during my journey, thank you!

**Pull request merged**